packages/dds/sequence/src/intervals/sequenceInterval.ts

```diff
@@ -205,14 +210,32 @@ export class SequenceInterval implements ISerializableInterval {
 	 * {@inheritDoc IInterval.compareStart}
 	 */
 	public compareStart(b: SequenceInterval) {
-		return compareReferencePositions(this.start, b.start);
+		const dist = compareReferencePositions(this.start, b.start);
+		if (dist !== 0 || this.startSide === b.startSide) {
+			return dist;
+		}
+
+		if (this.startSide === Side.Before) {
+			return 1;
+		}
+
+		return -1;
 	}
 
 	/**
 	 * {@inheritDoc IInterval.compareEnd}
 	 */
-	public compareEnd(b: SequenceInterval) {
-		return compareReferencePositions(this.end, b.end);
+	public compareEnd(b: SequenceInterval): number {
+		const dist = compareReferencePositions(this.end, b.end);
+		if (dist !== 0 || this.endSide === b.endSide) {
```

Comment:
not a big deal, but might be a little more idiomatic to write `function compareSides(a: Side, b: Side): number` and rewrite this bit to be standard lexicographical sort.

Reply:
seems worth it IMO to remove the duplication with compareStart.